consensus/hotstuff/validator/validator.go

```diff
@@ -113,6 +120,12 @@ func (v *Validator) ValidateProposal(proposal *model.Proposal) error {
 		return model.ErrUnverifiableBlock
 	}
 
+	// check validity of block timestamp using parent's timestamp
+	err = v.timestampValidator.Validate(parent.Timestamp, block.Timestamp)
+	if err != nil {
+		return newInvalidBlockError(block, err)
```

Comment:
I think we should check for the expected error type here:
 ```suggestion
		if !model.IsInvalidBlockTimestampError(err) {
			return fmt.Errorf("validating block's time stamp failed with unexpected error: %w", err)
		}
		return newInvalidBlockError(block, err)
```